### DISCUSSION TITLE
Example from load to display

### DISCUSSION THREAD

**Discussion by series0** [elimintz/justpy#411] [General]
New python developer here. Apologies.  Plenty of C experience.

Although I have found a number of examples here and read much of the documentation, no example I mimic allows me to simply load my pandas mysql dataframe up and display it on a webpage.  

I discovered the comment that I should open my local browser to see it and for the simple examples I was able to see it, but how would one simply open a browser in code (rather than have to do that manually).  I tried import Webbrowser, webbrowser.open(url) but it didnt work and placement of said code seems confusing to me.

When I load my dataframe it displays properly in Pandagui, but not in the justpy ag_grid.

I want a custom gui and grid I can modify on-the-fly and at initial configuration, both of which make pandagui not ideal.  The ag_grid (and later Quasar) as billed seem ideal.  

Is there a single example of load and display all the way through?  Also it seems the call to jp.justpy halts/traps execution.  How is that surrendered back to the code stream?  (Hope that question makes sense)

Finally, can the function called by the jp.justpy(function) be parameterized like 

def somefunc(paramlist)
    ...

jp.justpy(somefunc(paramlist))
print("Yo, I want this line to execute also so can the GUI be entered and exited multiple times?")

**Comment by elimintz** [comment]
It is difficult to give you a comprehensive answer because you need the basic concepts explained here: https://realpython.com/async-io-python/
Basically, you would run justpy like in all the examples, and then have another program or script that your users activate which opens a web browser with the appropriate url.

**Comment by series0** [comment]
OK, so, been through all the basic tutorials, and definitely got the pandas/ag_grid/mysql single web page component to work.  Yay!  

So, now software architecture comes to mind.  

I am seeing a lot of component level examples and that is great.  Aspects of most of these examples are relevant and useful.  Thank you!  The clear way everything is laid out is also very helpful.  

Now I want to go crazy creating use case components, but the debate I face immediately is ... is there a justpy way to clump them all together into a site?  I'm still too new at this to see the how of it.  I'm sure someone has done that ... so I am after pointers and ideas.

**Comment by elimintz** [comment]
Not sure what you are asking. There is no conceptual difference between a site with one component or a site with many components. Are you asking how to divide the project into several files? If so, just define the components in another file and import them to your main file.
